group_membership.tf

```
@@ -0,0 +1,12 @@
+# Put provisioner users in the appropriate group
+resource "aws_iam_user_group_membership" "provisioner" {
+  provider = aws.users
+
+  for_each = { for username, attributes in var.users : username => "" if contains(attributes["roles"], "provisioner") }
```

Does something like this work?
```suggestion
  for_each = toset([ for username, attributes in var.users : username if contains(attributes["roles"], "provisioner") ])
```